crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Generic.hs

```diff
@@ -681,6 +683,14 @@ readMemInvalidate sym w end mop (LLVMPointer blk off) tp d msg sz readPrev =
                      | otherwise =
                          symbolicRangeLoad pref tp
             evalMuxValueCtor sym w end mop varFn subFn mux0
+  where
+    readInRange :: ReadMem sym (PartLLVMVal sym)
+    readInRange
+      | laxLoadsAndStores ?memOpts &&
+        indeterminateLoadBehavior ?memOpts == UnstableSymbolic
+      = liftIO (Partial.totalLLVMVal sym <$> freshLLVMVal sym tp)
```

Comment:
I don't think this is the right storage type.  You need to use the `tp'` coming from the `InRange` constructor which tells you what potentially smaller write actually overlaps with the invalidate record.

Reply:
Oops, good catch! Fixed.